src/cgame/cg_main.cpp

```diff
@@ -1129,11 +1129,11 @@ static void GenerateNavmeshes()
 		return;
 	}
 
+	const std::string label = "Navmesh";
 	const std::string message = _("Generating bot navigation meshes");
 	cg.navmeshLoadingFraction = 0;
 	cg.loadingNavmesh = true;
-	Q_strncpyz( cg.loadingText, message.c_str(), sizeof(cg.loadingText) );
-	trap_UpdateScreen();
+	CG_UpdateLoadingProgress( LOAD_START, label.c_str(), message.c_str() );
```

Comment:
This commit seems like an awfully hacky way to make the code one line shorter. Setting a bogus value to cg.loadingFraction and emitting an  incorrect debug log message.

Reply:
Unrelated to this commit or comment, but could it be possible to start duplicating (yes, I know) interfaces to no longer use `char const*` but instead `std::string const&`? Thoses `.c_str()` that pollute the code are getting a bit annoying.

Reply:
We have StringRef for such interfaces. The point of StringRef is to make function arguments that can accept either std::string or char* without any .c_str() or inefficient copying.

Reply:
well, there's no copy with const references. But, I should take a look at this myself, thanks for info.

Reply:
Yes there is, if you put char* into a std::string argument then you are copying

Reply:
ofc

Reply:
> This commit seems like an awfully hacky way to make the code one line shorter. Setting a bogus value to cg.loadingFraction and emitting an incorrect debug log message.

Ah yes, I totally forgot I removed the printing of the log message because i was asked to remove it.

When I revamped this function when reimplementing map load message, I initially revamped it in order to be reusable, but this reusability was removed before merging so now the log message is meaningless when reusing the function.

Reply:
That code was totally rewritten so the thread is obsolete. New remarks should be done in another thread.